### PR TITLE
Enable containerDefinitions portMappings to use target_groups container_ports

### PR DESCRIPTION
**Pull request by networkprogrammer**
# Description

This commit enables the module to prefer the `target_groups` list of `container_port` and `protocol` over `task_container_port`.

If the target_groups does not have port and protocol specified, current behavior continues.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -168,6 +175,10 @@ resource "aws_ecs_task_definition" "task" {
   },
   %{~endif}
   "essential": true,
+
```
Comment (josh-onchain) [inline main.tf:178]: Remove whitespace :)

**Review by josh-onchain — APPROVED** [review]
Tested and works as expected gj!

**Review by josh-onchain — APPROVED** [review]

**Review by Ohid25 — APPROVED** [review]
LGTM 👍

**Comment by Ohid25** [comment]
Thanks for the contribution @networkprogrammer and the peer review @josh-onchain 🙌  - This is now released as [6.5.0](https://github.com/umotif-public/terraform-aws-ecs-fargate/releases/tag/6.5.0)!

**Pull request merged**